### PR TITLE
fix(AESND, ASND): :bug: AESND Aligns buffer. Callbacks should be right now

### PR DESCRIPTION
**Pull request by ProfElements**
make dsp time get use duration. 
Aesnd now allocates and aligns it buffers right.
callback should nwo be implemented correctly

ASND still needs to properly allocate it buffers.

### CODE REVIEW

File: src/aesnd.rs [hunk 1]
```diff
@@ -152,3 +159,9 @@ impl Aesnd {
         } 
     }
 }
+
+struct Align32<T>(pub T);
+
+fn alloc_sound_buffer(buffer: &[u8]) -> Align32<Vec<u8>> {
+    Align32((*buffer).to_vec())
```
Comment (1011X) [inline src/aesnd.rs:166]: nothing here guarantees that the buffer will be 32-byte aligned when allocated. check the source of `gx::Fifo::with_size()` to see how to do it.

or i can fix it later on, up to you.

**Review by 1011X — COMMENTED** [review]
can confirm that callback implementation seems sound. however, the buffers are not properly aligned. the implementation of `gx::Fifo::with_size()` can be used as a reference for how to do it if needed.

File: src/aesnd.rs (hunk 1 above)
Comment (ProfElements) [reply src/aesnd.rs:166]: I don't get what you mean? the repr on align32 must make the [u8] it takes aligned and padded to 32 bytes??

Comment (1011X) [reply src/aesnd.rs:166]: ... what repr? there's no repr attribute. even if there was, it wouldn't do what you need it to.

Comment (ProfElements) [reply src/aesnd.rs:166]: I see what I did! I used #[repr(C, align(32))] orginnaly in align32 but it isnt there anymore

**Pull request merged**